### PR TITLE
Remove explicit node version for workbench

### PR DESCRIPTION
**Pull request by joshuali925**
Signed-off-by: Joshua Li <joshuali925@gmail.com>

### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/sql-workbench-test-and-build-workflow.yml [hunk 1]
```diff
@@ -24,10 +24,16 @@ jobs:
           ref: ${{ env.OPENSEARCH_VERSION }}
           path: sql/OpenSearch-Dashboards
 
+      - name: Get node and yarn versions
+        id: versions_step
+        run: |
+          echo "::set-output name=node_version::$(node -p "(require('./OpenSearch-Dashboards/package.json').engines.node).match(/[.0-9]+/)[0]")"
+          echo "::set-output name=yarn_version::$(node -p "(require('./OpenSearch-Dashboards/package.json').engines.yarn).match(/[.0-9]+/)[0]")"
+
       - name: Setup Node
```
Comment (tmarkley) [inline .github/workflows/sql-workbench-test-and-build-workflow.yml:33]: Can you use the Dashboards `.nvmrc` file? Similar to what we do in Dashboards: https://github.com/tmarkley/OpenSearch-Dashboards/blob/c67b31e732995207ae432f24b393c902ec326b96/.github/workflows/build_and_test_workflow.yml#L32-L36

Comment (joshuali925) [reply .github/workflows/sql-workbench-test-and-build-workflow.yml:27]: sure, updated

File: .github/workflows/sql-workbench-test-and-build-workflow.yml [hunk 2]
```diff
@@ -25,9 +25,10 @@ jobs:
           path: sql/OpenSearch-Dashboards
 
       - name: Setup Node
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
-          node-version: '14.18.2'
+          node-version-file: ".nvmrc"
```
Comment (tmarkley) [inline .github/workflows/sql-workbench-test-and-build-workflow.yml:30]: I think you have to point to the OpenSearch Dashboards directory to get this to work.

Comment (tmarkley) [reply .github/workflows/sql-workbench-test-and-build-workflow.yml]: You beat me to it :)

**Review by mengweieric — APPROVED** [review]

**Review by kavithacm — APPROVED** [review]

**Pull request merged**